benchexec/outputhandler.py

```diff
@@ -308,9 +309,17 @@ def output_before_run_set(self, runSet):
         for run in runSet.runs:
             run.resultline = self.format_sourcefile_name(run.identifier, runSet)
 
+            adjusted_sourcefiles = [util.relative_path(s, xml_file_name) for s in run.sourcefiles]
+            if adjusted_sourcefiles:
+                adjusted_identifier = util.relative_path(run.identifier, xml_file_name)
+            else:
+                # If no source files exist the task doesn't point to any file that could be downloaded.
+                # In this case, the name doesn't have to be adjusted because it's no path.
+                adjusted_identifier = run.identifier
+
         # prepare XML structure for each run and runSet
             run.xml = ET.Element("run",
-                                 {"name": run.identifier, "files": "[" + ", ".join(run.sourcefiles) + "]"})
+                                 {"name": adjusted_identifier, "files": "[" + ", ".join(adjusted_sourcefiles) + "]"})
```

Comment:
Could you please remove the attribute for files completely for `<withoutfile>` runs instead of using `[]` as content? The format of this attribute is pretty ugly anyway (we will probably change this), so better avoid using it completely where possible.